libs/generic-view/ui/src/lib/interactive/input/search-input.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import React, { useEffect, useId } from "react"
+import { APIFC, IconType } from "generic-view/utils"
+import { withConfig } from "../../utils/with-config"
+import { withData } from "../../utils/with-data"
+import styled, { css } from "styled-components"
+import { IconButton } from "../../shared/button"
+import Icon from "../../icon/icon"
+import { useFormContext } from "react-hook-form"
+
+interface Data {
+  value: string
+}
+
+interface Config {
+  name: string
+  label: string
+}
+
+export const SearchInput: APIFC<Data, Config> = ({ data, config }) => {
+  const id = useId()
+  const { register, watch, setValue } = useFormContext()
+  const value = (watch(config!.name) as string) || ""
+
+  const clear = () => {
+    setValue(config!.name, "")
```

Comment:
⚪ (question)

Is using `!` with `config.name` necessary, and is there a safer alternative? Concerned about potential crashes

Reply:
This is because of `config` being optional (only as a type), while in fact it isn't. External API for generic views will prevent this but not placing a "name" or even a "config" should actually crash the app as it's just an error.

However, I'll add a topic for our front-end meeting so maybe we can discuss better solution from TypeScript's perspective.